database/db-driver/postgresql/db-query-tsj

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+# db-query-tsj -- Runs SQL against PostgreSQL and formats output rows in TSJ
+#
+# $ db-query-tsj SQL COLUMN_TYPE...
+##
+import psycopg2, psycopg2.extras, ujson
+import sys, subprocess, os, re
+
+# get arguments from command-line
+def usage(*msg):
+    err = subprocess.call(["usage", sys.argv[0]] + list(msg))
+    if err != 0: sys.exit(err)
+if not len(sys.argv) > 1: usage("Missing SQL")
+sql = sys.argv[1]
+if not len(sys.argv) > 2: usage("Missing COLUMN_TYPEs")
+column_types = sys.argv[2:]
+
+# write TSJ to stdout
+tsj_output = sys.stdout
+
+
+# SQL types that are already formatted in JSON (to bypass ujson.dump)
+TYPES_TO_PASS_THRU = [ "json" ]
+
+# PostgreSQL date/time types adapted by psycopg that need special formatting
+# See: http://initd.org/psycopg/docs/usage.html#adapt-date
+# See: https://www.postgresql.org/docs/current/static/datatype-datetime.html
+TYPES_DATETIME_PATTERNS = [re.compile(regex) for regex in (
+        r'date',
+        r'interval(?:|.+)',
+        r'timestamp\s*(?:\(\d\))?(?:\s*with(?:out)?\s+time\s+zone)?',
+        )]
+
+# generate code to output columns of each row based on their types
+# See: http://lucumr.pocoo.org/2011/2/1/exec-in-python/
+stmts = []
+for i,ty in enumerate(column_types):
+    ty = ty.lower()
+    if i > 0: # separate each column by TAB
+        stmts.append('tsj_output.write("\\t")')
+    if ty in TYPES_TO_PASS_THRU:
+        # passthru JSON compatible columns
+        stmts.append('tsj_output.write(columns[%d])' % i)
+    elif any(pattern.match(ty) for pattern in TYPES_DATETIME_PATTERNS):
+        # ujson needs us to format datetime
+        stmts.append('ujson.dump(columns[%d].isoformat(), tsj_output)' % i)
+    else:  # dump in JSON except a few
+        stmts.append('ujson.dump(columns[%d], tsj_output)' % i)
+stmts.append('tsj_output.write("\\n")')
+exec compile("def write_tsj_output(columns):\n" + "\n".join("  " + s for s in stmts),
+        "%s codegen" % sys.argv[0], "exec")
+
+# tell psycopg2 to actually skip parsing any JSON
+psycopg2.extras.register_default_json(loads=lambda x: x)
+# XXX or use ujson.loads if you really must parse it
+#psycopg2.extras.register_default_json(loads=ujson.loads)
+
+# use pyscopg2 to access the database
+conn = psycopg2.connect(
```

Comment:
Yeah, a separate package that we install in our virtualenv sounds great. I agree that we don't need to do anything special w/ .pyc's here.

When you say "CLI generator" I think of two different things:

1. For exposing executable commands from an installed Python package, consider using `entry_points`. Here's [a general overview](https://chriswarrick.com/blog/2014/09/15/python-apps-the-right-way-entry_points-and-scripts/) and here's [an example in memex](https://github.com/LatticeData/memex/tree/71b8ccb0730392c7ae442b44413eb6dae6ec3c69/queryjson). 

2. For parsing arguments, [`argparse`](https://docs.python.org/3/library/argparse.html) is the most common and straightfoward. When I have a really complex interface with lots of options I tend to use [`docopt`](https://github.com/docopt/docopt/blob/master/examples/arguments_example.py).

Let me know if I misunderstood and you were actually asking about something else.